matisse/src/main/java/com/zhihu/matisse/internal/ui/widget/RoundedRectangleImageView.java

```diff
@@ -16,42 +16,98 @@
 package com.zhihu.matisse.internal.ui.widget;
 
 import android.content.Context;
+import android.content.res.Configuration;
+import android.graphics.Bitmap;
 import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
 import android.graphics.Path;
+import android.graphics.PorterDuff;
+import android.graphics.PorterDuffXfermode;
 import android.graphics.RectF;
+import android.graphics.drawable.Drawable;
 import android.support.v7.widget.AppCompatImageView;
 import android.util.AttributeSet;
+import android.util.TypedValue;
 
 public class RoundedRectangleImageView extends AppCompatImageView {
```

Comment:
Eh.., but I tested it with a 10dp radius, it sometimes flashes with a unclipped image. And drawing a clipped image on every `onDraw` call may be expensive, but it really doesn't matter much 😄